src/Pack/Config/Environment/Variable.idr

```diff
@@ -0,0 +1,79 @@
+module Pack.Config.Environment.Variable
+
+import public Data.FilePath.File
+import Pack.Config.Types
+
+--------------------------------------------------------------------------------
+---         Types
+--------------------------------------------------------------------------------
+
+public export %inline
+DirList : Type
+DirList = List (Path Abs)
+
+export
+data EnvVar : Type where
+  MkEnvVar : (name : String) -> Interpolation a => (value : a) -> EnvVar
+
+--------------------------------------------------------------------------------
+---         Interpolation
+--------------------------------------------------------------------------------
+
+||| Interpolate a list of directories as a colon-separated string
+export
+Interpolation DirList where
+  interpolate = fastConcat . intersperse ":" . map interpolate
+
+||| Interpolate an environment variable for use in a shell
+export
+Interpolation EnvVar where
+  interpolate (MkEnvVar var val) = "\{var}=\{val}"
+
+--------------------------------------------------------------------------------
+--          Environment Variables
+--------------------------------------------------------------------------------
+
+-- The official documentation for Idris 2 compiler environment variables:
+-- https://github.com/idris-lang/Idris2/blob/main/docs/source/reference/envvars.rst
+
+||| Code generation backend used by the Idris 2 compiler
+public export %inline
+IdrisCodegenVar : Codegen -> EnvVar
+IdrisCodegenVar = MkEnvVar "IDRIS2_CG"
+
+||| Path to the Scheme executable used by the Scheme backend
+public export %inline
+SchemeVar : FilePath -> EnvVar
+SchemeVar = MkEnvVar "SCHEME"
+
+||| Path to the Idris 2 boot file, used by Idris 2 Makefile
+public export %inline
+IdrisBootVar : File Abs -> EnvVar
+IdrisBootVar = MkEnvVar "IDRIS2_BOOT"
```

Comment:
I am confused by the difference in argument types for `SchemeVar` and `IdrisBootVar`. But right now, it matches the rest of the code. https://github.com/stefan-hoeck/idris2-pack/blob/a62f0d7601a2c4e89a3cf19fa7d8f1b16ad080b0/src/Pack/Config/Types.idr#L178-L179 https://github.com/stefan-hoeck/idris2-pack/blob/a62f0d7601a2c4e89a3cf19fa7d8f1b16ad080b0/src/Pack/Config/Environment.idr#L146-L147 What do you think about this? Do we need casts here?